test/logger_acc.erl

```diff
@@ -0,0 +1,98 @@
+%% @doc A logger report handler that be used to capture expected errors in
+%% tests. The current error report handlers are disabled during the execution
+%% of a function. Afterwards, they are restored and the errors that occered are
+%% returned along with the return value of the fun.
+-module(logger_acc).
+
+%% Public API
+-export([capture/1]).
+
+%% @doc Executes `Fun' and captures all logged errors returns as well as
+%% uncaught errors in `Fun'.
+-spec capture(fun (() -> ResultOfFun)) ->
+    {ok, ResultOfFun, AccumulatedErrors} |
+    {throw | error | exit, Reason, Trace, AccumulatedErrors}
+  when ResultOfFun :: term(),
+       Reason :: term(),
+       Trace :: list(),
+       AccumulatedErrors :: [{error|warning_msg|info_msg, string()} |
+                             {error_report|warning_report|info_report, term()}].
+capture(Fun) when is_function(Fun, 0) ->
+    Tag = make_ref(),
+    Self = self(),
+    AccPid = spawn_link(fun() -> log_acc_loop(Tag, Self) end),
+    logger:add_primary_filter(?MODULE, {fun(Event, _)-> AccPid ! {Tag, Event}, stop end, undefined}),
+    try
+        Fun()
+    of
+        Result ->
+            Events = flush_logs(AccPid, Tag),
+            {ok, Result, Events}
+    catch
+        Class:Error:Stacktrace ->
+            Events = flush_logs(AccPid, Tag),
+            {Class, Error, Stacktrace, Events}
+    after
+        ok = logger:remove_primary_filter(?MODULE)
+    end.
+
+flush_logs(AccPid, Tag) ->
+    AccMon = monitor(process, AccPid),
+    AccPid ! {Tag, flush},
+    receive
+        {Tag, Events} ->
+            demonitor(AccMon, [flush]),
+            Events;
+        {'DOWN', AccMon, process, AccPid, Reason} ->
+            error({accumulator_process, Reason})
+    end.
+
+log_acc_loop(Tag, Parent) ->
+    log_acc_loop(Tag, Parent, open, []).
+
+log_acc_loop(Tag, Parent, Status, Acc) ->
+    Timeout = case Status of
+        open -> infinity;
+	closing -> 1000
+    end,
```

Comment:
When I wrote that code 10-ish years ago, I didn't know the proper indentation of Erlang code. :)

Are those tabs btw? Please use spaces.

```suggestion
    Timeout = case Status of
                  open -> infinity;
                  closing -> 1000
              end,
```

Reply:
Indeed... they keep sneaking in :sweat_smile:

Reply:
> When I wrote that code 10-ish years ago, I didn't know the proper indentation of Erlang code. :)

I'm going back and forth between different styles, and I see that I even used 2 different ones in this very function :sweat_smile: I think the style used by OTP itself is to align the `case` and `end` in the same column, and to indent with 4 spaces starting from that column.

Reply:
All fixed in last commit.

Reply:
Yes. I like this style, and it's what you get if you're using emacs. (I do, nowdays. It's also possible to run it in CI with erlfmt.)

Reply:
I'm sticking to vim :blush: